realm/realm-library/src/main/java/io/realm/coroutines/RealmFlowFactory.kt

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.coroutines
+
+import io.realm.*
+import io.realm.annotations.Beta
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.flowOf
+
+/**
+ * Flow factory used by Realm by default.
```

Comment:
What actually happens in our Javadoc? Is this class even listed there? If not, that is a problem, that might prevent us from implementing this in Kotlin. 🤔

Reply:
I had to add an exception to exclude Kotlin files from being javadoc'ed in the build.gradle script since the mere existence of this class was causing the javadoc task to fail. I think it is totally ignored at the moment. But it is a valid concern. I don't think javadoc can ever document Kotlin classes, it can only be done with dokka, but then I'm not sure how to glue the two outputs together...

Reply:
It seems it might be possible to generate the javadoc using dokka instead: https://blog.jetbrains.com/kotlin/2020/08/dokka-preview-based-on-kotlin-1-4-0-rc/

Reply:
I am unsure about being able to write this class in Java. The file containing the `ProducerScope.awaitClose` extension function doesn't have the `@file:JvmName` that would make it visible from Java, and so far it seems impossible to call it using a `ProducerScope` instance from java 😞

Reply:
Will add an internal Kotlin class that will be called from a Java class that implements the flow interface. This way we will not have issues with javadoc.